builtin/providers/openstack/resource_openstack_fw_policy_v1.go

```diff
@@ -0,0 +1,225 @@
+)
+
+func resourceFWPolicyV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceFWPolicyV1Create,
+		Read:   resourceFWPolicyV1Read,
+		Update: resourceFWPolicyV1Update,
+		Delete: resourceFWPolicyV1Delete,
+
+		Schema: map[string]*schema.Schema{
+			"region": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				DefaultFunc: envDefaultFunc("OS_REGION_NAME"),
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"audited": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"shared": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"tenant_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"rules": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+		},
+	}
+}
+
+func resourceFWPolicyV1Create(d *schema.ResourceData, meta interface{}) error {
+
+	config := meta.(*Config)
+	networkingClient, err := config.networkingV2Client(d.Get("region").(string))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack networking client: %s", err)
+	}
+
+	v := d.Get("rules").(*schema.Set)
+
+	log.Printf("[DEBUG] Rules found : %#v", v)
+	log.Printf("[DEBUG] Rules count : %d", v.Len())
+
+	rules := make([]string, v.Len())
+	for i, v := range v.List() {
+		rules[i] = v.(string)
+	}
+
+	audited := d.Get("audited").(bool)
+	shared := d.Get("shared").(bool)
+
+	opts := policies.CreateOpts{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		Audited:     &audited,
+		Shared:      &shared,
+		TenantID:    d.Get("tenant_id").(string),
+		Rules:       rules,
+	}
+
+	log.Printf("[DEBUG] Create firewall policy: %#v", opts)
+
+	policy, err := policies.Create(networkingClient, opts).Extract()
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Firewall policy craeted: %#v", policy)
+
+	d.SetId(policy.ID)
+
+	return resourceFWPolicyV1Read(d, meta)
+}
+
+func resourceFWPolicyV1Read(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] Retrieve information about firewall policy: %s", d.Id())
+
+	config := meta.(*Config)
+	networkingClient, err := config.networkingV2Client(d.Get("region").(string))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack networking client: %s", err)
+	}
+
+	policy, err := policies.Get(networkingClient, d.Id()).Extract()
+
+	if err != nil {
+		return CheckDeleted(d, err, "LB pool")
+	}
+
+	if t, exists := d.GetOk("name"); exists && t != "" {
+		d.Set("name", policy.Name)
+	} else {
+		d.Set("name", "")
+	}
```

Comment:
Similar to my earlier recommendation, I think these can just be simplified to `d.Set("name", policy.Name)` etc etc without the guards.